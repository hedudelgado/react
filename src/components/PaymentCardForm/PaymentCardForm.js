import React from 'react';
import {
  array,
  bool,
  string,
  object,
  func,
} from 'prop-types';
import Typography from '../../layout/Typography';
import BaseStyle from '../../layout/BaseStyle';
import Button from '../../layout/Button';
import Input from '../../layout/Input';
import Select from '../../layout/Select';
import RadioGroup, { RadioItem } from '../../layout/RadioGroup';
import Checkbox from '../../layout/Checkbox';
import CardNumberInput from '../../layout/CardNumberInput';
import AddressForm from '../AddressForm';

const PaymentCardForm = ({
  dictionary,
  countries,
  paymentCards,
  handleCardNumberBlur,
  handleSubmit,
  setFieldTouched,
  setFieldError,
  setFieldValue,
  handleBlur,
  handleChange,
  cancel,
  errors,
  touched,
  values,
  cardType,
  domain,
  postCode,
  disableSelectDefaults,
  showExpiryDateError,
}) => {
  const months = process.env.REACT_APP_MONTHS
    .split(',')
    .map(month => <option key={month} value={parseInt(month, 10)}>{month}</option>);

  const currentYear = new Date().getFullYear();

  const years = [...Array(parseInt(process.env.REACT_APP_NUM_YEARS, 10))]
    .map((item, index) => {
      const year = index + currentYear;
      return (<option key={year} value={year}>{year}</option>);
    });

  const { usePersonalAddress, prePayment } = values;
  return (
    <form
      data-test="payment-card-form"
      name="payment-card-form"
      onSubmit={handleSubmit}
    >
      <Input
        onBlur={handleBlur}
        onChange={handleChange}
        type="text"
        placeholder={dictionary['payment.add.name']}
        id="cardHolderName"
        name="cardHolderName"
        mb="mb2"
        error={dictionary[errors.cardHolderName]}
        touched={touched.cardHolderName}
      />
      <CardNumberInput
        onChange={handleChange}
        type="tel"
        name="cardNumber"
        placeholder={dictionary['payment.add.number']}
        error={dictionary[errors.cardNumber]}
        touched={touched.cardNumber}
        card={paymentCards[cardType] && {
          src: domain + paymentCards[cardType].img,
          alt: paymentCards[cardType].name,
        }}
        onBlur={() => handleCardNumberBlur(
          values.cardNumber,
          setFieldTouched,
          setFieldError,
        )}
        sessioncamhide
      />
      <Typography size="s" weight="bold" className="pi-color-greyscale-darkest mb0 mt2 wb-push-half--bottom">
        {dictionary['payment.add.date']}
      </Typography>
      <BaseStyle className="clearfix wb-push-middle--bottom" component="div">
        <BaseStyle className="clearfix" component="div">
          <Select
            onBlur={handleBlur}
            onChange={handleChange}
            name="expiryMonth"
            id="expiryMonth"
            touched={touched.expiryYear && touched.expiryMonth}
            error={dictionary[errors.expiryMonth]}
            sessioncamhide
            halfWidth
            hideIcon
            hideMessage
          >
            <option key="month" value="" defaultValue disabled={disableSelectDefaults.month}>
              {dictionary['payment.placeholder.month']}
            </option>
            {months}
          </Select>
          <Select
            onBlur={handleBlur}
            onChange={handleChange}
            name="expiryYear"
            id="expiryYear"
            touched={touched.expiryYear && touched.expiryMonth}
            error={dictionary[errors.expiryYear]}
            sessioncamhide
            halfWidth
            hideMessage
          >
            <option key="year" value="" disabled={disableSelectDefaults.year}>
              {dictionary['payment.placeholder.year']}
            </option>
            {years}
          </Select>
        </BaseStyle>
        {showExpiryDateError && touched.expiryYear && touched.expiryMonth && (
          <Typography
            data-test="expiry-date-error"
            component="span"
            className="form-item-msg form-item-msg--error"
          >
            {dictionary[errors.expiryYear]}
          </Typography>
        )}
      </BaseStyle>
      <RadioGroup>
        <RadioItem
          field={{
            name: 'usePersonalAddress',
            value: usePersonalAddress,
            onBlur: handleBlur,
            onChange: handleChange,
          }}
          label={dictionary['payment.add.address.same']}
          name="usePersonalAddress"
          id="usePersonalAddress"
          value="true"
          subLabels={[postCode]}
          sessioncamhide
        />
        <RadioItem
          field={{
            name: 'usePersonalAddress',
            value: usePersonalAddress,
            onBlur: handleBlur,
            onChange: handleChange,
          }}
          label={dictionary['payment.add.address.different']}
          name="usePersonalAddress"
          id="useBillingAddress"
          value="false"
          sessioncamhide
        />
      </RadioGroup>
      {usePersonalAddress === 'false' && (
        <AddressForm
          dictionary={dictionary}
          countries={countries}
          values={values}
          setFieldValue={setFieldValue}
          errors={errors}
          touched={touched}
          handleBlur={handleBlur}
          handleChange={handleChange}
        />
      )}
      <Checkbox
        field={{
          name: 'prePayment',
          value: prePayment,
          onBlur: handleBlur,
          onChange: handleChange,
        }}
        name="prePayment"
        id="prePayment"
        label={dictionary['payment.add.prepay']}
        sessioncamhide
      />
      <Button
        className="wb-push--bottom"
        id="save-card"
        color="primary"
        type="submit"
      >
        {dictionary['payment.add.save']}
      </Button>
      <Button
        data-test="closeForm"
        variant="text"
        color="default"
        className="wb-text-btn--underline block wb-push--bottom"
        onClick={cancel}
      >
        {dictionary['payment.add.cancel']}
      </Button>
    </form>
  );
};

PaymentCardForm.propTypes = {
  dictionary: object.isRequired,
  countries: array.isRequired,
  errors: object.isRequired,
  touched: object.isRequired,
  values: object.isRequired,
  disableSelectDefaults: object.isRequired,
  paymentCards: object.isRequired,
  handleCardNumberBlur: func.isRequired,
  handleSubmit: func.isRequired,
  setFieldTouched: func.isRequired,
  setFieldError: func.isRequired,
  setFieldValue: func.isRequired,
  handleBlur: func.isRequired,
  cancel: func.isRequired,
  handleChange: func.isRequired,
  cardType: string.isRequired,
  domain: string.isRequired,
  postCode: string.isRequired,
  showExpiryDateError: bool.isRequired,
};

export default PaymentCardForm;
