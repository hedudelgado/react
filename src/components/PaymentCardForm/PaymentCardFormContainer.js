import React, { PureComponent } from 'react';
import { array, func, object } from 'prop-types';
import { Formik } from 'formik';
import { validateBin } from '../../utils/api';
import { DEFAULT_VIEW, PAYMENT_CARD_ERROR, PAYMENT_CARD_VIEW } from '../App/appActions';
import Typography from '../../layout/Typography';
import { getJsonItem } from '../../utils/session-storage';
import BaseStyle from '../../layout/BaseStyle';
import {
  validateCardNumber,
  validateExpiryDate,
  validateFullName,
  validateCompanyName,
  validateAddressLine,
  validatePostCode,
} from '../../utils/validations';
import Img from '../../layout/Img';
import Notification from '../../layout/Notification';
import PaymentCardForm from './PaymentCardForm';
import {
  BUSINESS,
  LEISURE,
  paymentCards,
} from '../../config.json';

const getDomain = () => getJsonItem('environment').domain;

const CardList = () => Object.keys(paymentCards)
  .map(card => (
    <Img
      key={paymentCards[card].name}
      src={getDomain() + paymentCards[card].img}
      label={paymentCards[card].name}
      className="mr1"
    />
  ));

export default class PaymentCardFormContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      cardType: '',
      showExpiryDateError: false,
      disableSelectDefaults: {
        month: false,
        year: false,
      },
    };

    this.cancel = this.cancel.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.validateExpiryDate = this.validateExpiryDate.bind(this);
    this.handleCardNumberBlur = this.handleCardNumberBlur.bind(this);
    this.validatePaymentCardForm = this.validatePaymentCardForm.bind(this);
  }

  handleSubmit(values, actions) {
    const {
      setNotificationMessage,
      setEditView,
      updateProfileStore,
      updateProfile,
    } = this.props;
    const payload = this.createPayload(values);

    updateProfile(payload)
      .then(() => {
        setNotificationMessage(PAYMENT_CARD_VIEW);
        setEditView(DEFAULT_VIEW);
        actions.setSubmitting(false);
        const { paymentPreference: { paymentCard: { cardNumber } } } = payload;
        payload.paymentPreference.paymentCard.cardNumber = `************${cardNumber.slice(cardNumber.length - 4)}`;
        updateProfileStore(payload);
      }).catch(() => {
        setNotificationMessage(PAYMENT_CARD_ERROR);
      });
  }

  createPayload(values) {
    const {
      cardNumber,
      cardHolderName,
      expiryMonth,
      expiryYear,
      prePayment,
      usePersonalAddress,
      line1,
      line2,
      line3,
      line4,
      type,
      postCode,
      countryCode,
      companyName,
    } = values;

    const { profile } = this.props;
    const { cardType } = this.state;

    const expiryDate = `${(expiryMonth < 10 ? '0' : '')}${expiryMonth}/${expiryYear % 2000}`;

    const payload = {
      ...profile,
      paymentPreference: {
        ...profile.paymentPreference,
        paymentCard: {
          cardNumber,
          cardHolderName,
          cardType,
          expiryDate,
        },
      },
      bookingPreference: {
        ...profile.bookingPreference,
        prepay: prePayment,
      },
    };

    if (usePersonalAddress === 'false') {
      payload.paymentPreference.paymentCard.billingAddress = {
        line1,
        line2,
        line3,
        line4,
        postCode,
        countryCode,
        type,
        companyName: type === BUSINESS ? companyName : null,
      };
    }
    return payload;
  }

  cancel() {
    const { setEditView, setNotificationMessage } = this.props;
    setEditView(DEFAULT_VIEW);
    setNotificationMessage(DEFAULT_VIEW);
  }

  handleCardNumberBlur(cardNumber, setFieldTouched, setFieldError) {
    const result = validateCardNumber(cardNumber);
    setFieldTouched('cardNumber', true, false);
    if (!result) {
      validateBin(cardNumber)
        .then(({ cardType }) => {
          this.setState({ cardType });
        });
    } else {
      setFieldError('cardNumber', result);
      this.setState({ cardType: '' });
    }
  }

  validateExpiryDate(year, month) {
    const { disableSelectDefaults } = this.state;
    const result = validateExpiryDate(year, month);

    this.setState({
      showExpiryDateError: !!result,
      disableSelectDefaults: {
        ...disableSelectDefaults,
        month: !!month,
        year: !!year,
      },
    });

    return result;
  }

  validatePaymentCardForm({
    cardNumber,
    cardHolderName,
    expiryMonth,
    expiryYear,
    line1,
    line2,
    line3,
    line4,
    postCode,
    countryCode,
    type,
    companyName,
    usePersonalAddress,
  }) {
    const errors = {};

    errors.cardHolderName = validateFullName(cardHolderName);
    errors.cardNumber = validateCardNumber(cardNumber);

    const errorExpiryDate = this.validateExpiryDate(expiryYear, expiryMonth);
    if (errorExpiryDate) {
      errors.expiryYear = errorExpiryDate;
      errors.expiryMonth = errorExpiryDate;
    }

    if (usePersonalAddress === 'false') {
      errors.postCode = validatePostCode(postCode, countryCode);

      if (type === 'BUSINESS') {
        errors.companyName = validateCompanyName(companyName);
      }

      errors.line1 = validateAddressLine(line1, true);
      errors.line2 = validateAddressLine(line2, false);
      errors.line3 = validateAddressLine(line3, false);
      errors.line4 = validateAddressLine(line4, false);
    }

    Object.keys(errors).forEach((error) => {
      if (errors[error] === undefined) delete errors[error];
    });

    return errors;
  }

  render() {
    const {
      dictionary,
      countries,
      profile: {
        contactDetail: {
          address: {
            postCode,
          },
        },
      },
      app: { notificationMessage },
    } = this.props;
    const {
      cardType,
      showExpiryDateError,
      disableSelectDefaults,
    } = this.state;

    return (
      <BaseStyle component="section" className="component-container">
        {notificationMessage === PAYMENT_CARD_ERROR && (
          <Notification data-test="error-notification" priority={2} variant="error" ariaLive="polite" show>
            {dictionary['generic.error']}
          </Notification>
        )}
        <Typography component="h4" className="pi-color-greyscale-darkest">
          {dictionary['payment.add.title']}
        </Typography>
        <BaseStyle data-test="payment-cards" component="div" className="mb2">
          <CardList paymentCards={paymentCards} domain={getDomain()} />
        </BaseStyle>
        <Formik
          data-test="payment-card-form-wrapper"
          onSubmit={this.handleSubmit}
          initialValues={{
            cardNumber: '',
            cardHolderName: '',
            expiryMonth: '',
            expiryYear: '',
            usePersonalAddress: 'true',
            prePayment: 'false',
            line1: '',
            line2: '',
            line3: '',
            line4: '',
            postCode: '',
            countryCode: 'GB',
            type: LEISURE,
            companyName: '',
          }}
          validate={this.validatePaymentCardForm}
          render={formikProps => (
            <PaymentCardForm
              {...formikProps}
              dictionary={dictionary}
              countries={countries}
              handleCardNumberBlur={this.handleCardNumberBlur}
              cancel={this.cancel}
              paymentCards={paymentCards}
              cardType={cardType}
              domain={getDomain()}
              postCode={postCode}
              showExpiryDateError={showExpiryDateError}
              disableSelectDefaults={disableSelectDefaults}
            />
          )}
        />
      </BaseStyle>
    );
  }
}

PaymentCardFormContainer.propTypes = {
  app: object.isRequired,
  dictionary: object.isRequired,
  setEditView: func.isRequired,
  profile: object.isRequired,
  setNotificationMessage: func.isRequired,
  updateProfileStore: func.isRequired,
  countries: array.isRequired,
  updateProfile: func.isRequired,
};
