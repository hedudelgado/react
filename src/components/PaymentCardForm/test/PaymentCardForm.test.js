import React from 'react';
import { mount } from 'enzyme';
import PaymentCardForm from '../PaymentCardForm';

describe('PaymentCardForm Component', () => {
  let wrapper;

  const props = {
    dictionary: {
      'payment.add.prepay': 'test',
    },
    paymentCards: {},
    errors: {},
    touched: {},
    values: {},
    countries: [],
    setFieldTouched: jest.fn(),
    setFieldError: jest.fn(),
    setFieldValue: jest.fn(),
    handleSubmit: jest.fn(),
    handleBlur: jest.fn(),
    handleChange: jest.fn(),
    cancel: jest.fn(),
    handleCardNumberBlur: jest.fn(),
    cardType: '',
    domain: '',
    postCode: '',
    showExpiryDateError: false,
    disableSelectDefaults: {},
  };

  beforeEach(() => {
    wrapper = mount(<PaymentCardForm {...props} />);
    jest.clearAllMocks();
  });

  describe('Renders Payment Card Form Component Elements', () => {
    it('should render the component', () => {
      expect(wrapper).toMatchSnapshot();
    });


    it('should show invalid expiry date error if users interacts with the fields and date is invalid', () => {
      wrapper.setProps({
        touched: {
          expiryMonth: true,
          expiryYear: true,
        },
        showExpiryDateError: true,
      });
      expect(wrapper).toContainExactlyOneMatchingElement('Typography[data-test="expiry-date-error"]');
    });
  });
});
