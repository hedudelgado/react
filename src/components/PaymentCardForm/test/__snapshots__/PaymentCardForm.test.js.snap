// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentCardForm Component Renders Payment Card Form Component Elements should render the component 1`] = `
<PaymentCardForm
  cancel={[MockFunction]}
  cardType=""
  countries={Array []}
  dictionary={
    Object {
      "payment.add.prepay": "test",
    }
  }
  disableSelectDefaults={Object {}}
  domain=""
  errors={Object {}}
  handleBlur={[MockFunction]}
  handleCardNumberBlur={[MockFunction]}
  handleChange={[MockFunction]}
  handleSubmit={[MockFunction]}
  paymentCards={Object {}}
  postCode=""
  setFieldError={[MockFunction]}
  setFieldTouched={[MockFunction]}
  setFieldValue={[MockFunction]}
  showExpiryDateError={false}
  touched={Object {}}
  values={Object {}}
>
  <form
    data-test="payment-card-form"
    name="payment-card-form"
    onSubmit={[MockFunction]}
  >
    <Input
      id="cardHolderName"
      mb="mb2"
      name="cardHolderName"
      onBlur={[MockFunction]}
      onChange={[MockFunction]}
      sessioncamhide={true}
      type="text"
    >
      <div
        className="wb-form-item mb2"
      >
        <div
          className="wb-form-item__field wb-form-item__field__full"
        >
          <input
            className="sessioncamhidetext wb-dark"
            id="cardHolderName"
            name="cardHolderName"
            onBlur={[MockFunction]}
            onChange={[MockFunction]}
            type="text"
          />
        </div>
      </div>
    </Input>
    <CardNumberInput
      card={Object {}}
      name="cardNumber"
      onBlur={[Function]}
      onChange={[MockFunction]}
      sessioncamhide={true}
      type="tel"
    >
      <div
        className="wb-form-item mb2"
      >
        <div
          className="wb-form-item__field wb-form-item__field__full"
        >
          <input
            className="sessioncamhidetext"
            name="cardNumber"
            onBlur={[Function]}
            onChange={[MockFunction]}
            type="tel"
          />
        </div>
      </div>
    </CardNumberInput>
    <Typography
      className="pi-color-greyscale-darkest mb0 mt2 wb-push-half--bottom"
      component="p"
      size="s"
      weight="bold"
    >
      <BaseStyle
        className="font-size--s wb-strong pi-color-greyscale-darkest mb0 mt2 wb-push-half--bottom"
        component="p"
      >
        <p
          className="font-size--s wb-strong pi-color-greyscale-darkest mb0 mt2 wb-push-half--bottom"
        />
      </BaseStyle>
    </Typography>
    <BaseStyle
      className="clearfix wb-push-middle--bottom"
      component="div"
    >
      <div
        className="clearfix wb-push-middle--bottom"
      >
        <BaseStyle
          className="clearfix"
          component="div"
        >
          <div
            className="clearfix"
          >
            <Select
              halfWidth={true}
              hideIcon={true}
              hideMessage={true}
              id="expiryMonth"
              name="expiryMonth"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              placeholder=""
              sessioncamhide={true}
            >
              <div
                className="wb-form-item col mr2"
              >
                <div
                  className="wb-form-item__field wb-form-item__field__half"
                >
                  <select
                    className="sessioncamhidetext"
                    id="expiryMonth"
                    name="expiryMonth"
                    onBlur={[MockFunction]}
                    onChange={[MockFunction]}
                  >
                    <option
                      defaultValue={true}
                      key="month"
                      value=""
                    />
                    <option
                      key="01"
                      value={1}
                    >
                      01
                    </option>
                    <option
                      key="02"
                      value={2}
                    >
                      02
                    </option>
                    <option
                      key="03"
                      value={3}
                    >
                      03
                    </option>
                    <option
                      key="04"
                      value={4}
                    >
                      04
                    </option>
                    <option
                      key="05"
                      value={5}
                    >
                      05
                    </option>
                    <option
                      key="06"
                      value={6}
                    >
                      06
                    </option>
                    <option
                      key="07"
                      value={7}
                    >
                      07
                    </option>
                    <option
                      key="08"
                      value={8}
                    >
                      08
                    </option>
                    <option
                      key="09"
                      value={9}
                    >
                      09
                    </option>
                    <option
                      key="10"
                      value={10}
                    >
                      10
                    </option>
                    <option
                      key="11"
                      value={11}
                    >
                      11
                    </option>
                    <option
                      key="12"
                      value={12}
                    >
                      12
                    </option>
                  </select>
                  <div
                    className="wb-icon_arrow-down"
                  />
                </div>
              </div>
            </Select>
            <Select
              halfWidth={true}
              hideIcon={false}
              hideMessage={true}
              id="expiryYear"
              name="expiryYear"
              onBlur={[MockFunction]}
              onChange={[MockFunction]}
              placeholder=""
              sessioncamhide={true}
            >
              <div
                className="wb-form-item col mr2"
              >
                <div
                  className="wb-form-item__field wb-form-item__field__half"
                >
                  <select
                    className="sessioncamhidetext"
                    id="expiryYear"
                    name="expiryYear"
                    onBlur={[MockFunction]}
                    onChange={[MockFunction]}
                  >
                    <option
                      key="year"
                      value=""
                    />
                    <option
                      key="2019"
                      value={2019}
                    >
                      2019
                    </option>
                    <option
                      key="2020"
                      value={2020}
                    >
                      2020
                    </option>
                    <option
                      key="2021"
                      value={2021}
                    >
                      2021
                    </option>
                    <option
                      key="2022"
                      value={2022}
                    >
                      2022
                    </option>
                    <option
                      key="2023"
                      value={2023}
                    >
                      2023
                    </option>
                    <option
                      key="2024"
                      value={2024}
                    >
                      2024
                    </option>
                    <option
                      key="2025"
                      value={2025}
                    >
                      2025
                    </option>
                    <option
                      key="2026"
                      value={2026}
                    >
                      2026
                    </option>
                    <option
                      key="2027"
                      value={2027}
                    >
                      2027
                    </option>
                    <option
                      key="2028"
                      value={2028}
                    >
                      2028
                    </option>
                  </select>
                  <div
                    className="wb-icon_arrow-down"
                  />
                </div>
              </div>
            </Select>
          </div>
        </BaseStyle>
      </div>
    </BaseStyle>
    <RadioGroup>
      <div
        className="wb-form-item fixed-size mb3 wb-form-item--clean max-fit"
      >
        <ul
          className="wb-radio__group"
        >
          <RadioItem
            field={
              Object {
                "name": "usePersonalAddress",
                "onBlur": [MockFunction],
                "onChange": [MockFunction],
                "value": undefined,
              }
            }
            id="usePersonalAddress"
            name="usePersonalAddress"
            sessioncamhide={true}
            subLabels={
              Array [
                "",
              ]
            }
            value="true"
          >
            <li
              className="wb-radio__inner"
            >
              <label
                className="wb-radio__label"
                htmlFor="usePersonalAddress"
              >
                <input
                  checked={false}
                  className="sessioncamhidetext"
                  id="usePersonalAddress"
                  name="usePersonalAddress"
                  onBlur={[MockFunction]}
                  onChange={[MockFunction]}
                  type="radio"
                  value="true"
                />
                <span
                  className="wb-radio__label__inner"
                >
                  <span
                    className="wb-radio__label__inner__title"
                  />
                  <span
                    className="wb-radio__label__inner-info"
                    key=""
                  />
                </span>
              </label>
            </li>
          </RadioItem>
          <RadioItem
            field={
              Object {
                "name": "usePersonalAddress",
                "onBlur": [MockFunction],
                "onChange": [MockFunction],
                "value": undefined,
              }
            }
            id="useBillingAddress"
            name="usePersonalAddress"
            sessioncamhide={true}
            subLabels={Array []}
            value="false"
          >
            <li
              className="wb-radio__inner"
            >
              <label
                className="wb-radio__label"
                htmlFor="useBillingAddress"
              >
                <input
                  checked={false}
                  className="sessioncamhidetext"
                  id="useBillingAddress"
                  name="usePersonalAddress"
                  onBlur={[MockFunction]}
                  onChange={[MockFunction]}
                  type="radio"
                  value="false"
                />
                <span
                  className="wb-radio__label__inner"
                >
                  <span
                    className="wb-radio__label__inner__title"
                  />
                </span>
              </label>
            </li>
          </RadioItem>
        </ul>
      </div>
    </RadioGroup>
    <Checkbox
      field={
        Object {
          "name": "prePayment",
          "onBlur": [MockFunction],
          "onChange": [MockFunction],
          "value": undefined,
        }
      }
      id="prePayment"
      label="test"
      name="prePayment"
      sessioncamhide={true}
    >
      <div
        className="wb-form-item wb-checkbox wb-checkbox__white wb-form-item--clean mb3"
      >
        <input
          className="sessioncamhidetext"
          id="prePayment"
          name="prePayment"
          onBlur={[MockFunction]}
          onChange={[MockFunction]}
          type="checkbox"
        />
        <label
          htmlFor="prePayment"
        >
          test
        </label>
      </div>
    </Checkbox>
    <Button
      className="wb-push--bottom"
      color="primary"
      fullWidth={false}
      href={null}
      id="save-card"
      onClick={[Function]}
      submit={false}
      type="submit"
      variant="button"
    >
      <button
        className="wb-btn wb-btn--primary wb-btn--small wb-push--bottom"
        onClick={[Function]}
        type="submit"
      />
    </Button>
    <Button
      className="wb-text-btn--underline block wb-push--bottom"
      color="default"
      data-test="closeForm"
      fullWidth={false}
      href={null}
      onClick={[MockFunction]}
      submit={false}
      variant="text"
    >
      <button
        className="wb-text-btn wb-text-btn--default wb-text-btn--underline block wb-push--bottom"
        onClick={[MockFunction]}
        type="button"
      />
    </Button>
  </form>
</PaymentCardForm>
`;
